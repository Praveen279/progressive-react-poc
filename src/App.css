body {
  background: #CCC;
}

.channels-wrapper {
  margin: 20px;
}

.accordian-child-class {
  border-left: 1px solid #FFF;
  min-height: 100px;
  margin: 20px;
}

.c-item-container {
  background: #FFF;
  width: 120px;
  padding: 10px;
  border-radius: 5px;
  display: inline-block;
  margin-right: 20px;
}

.c-item-container-selected {
  background: #00BD73;
}

.c-label {
  font-size: 12px;
  text-transform: uppercase;
  word-wrap: break-word;
  padding-bottom: 5px;
}

.c-content {
  font-size: 10px;
  word-wrap: break-word;
}

.channels-accordian {
  margin: 20px 30px;
  min-height: 180px;
}
