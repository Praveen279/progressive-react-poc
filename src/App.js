import React from 'react';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { client } from './constants';

const ADD_TODO = gql`
  mutation AddTodo($type: String!) {
    addTodo(type: $type) {
      id
      type
    }
  }
`;

class App extends React.Component {

  state = {
    rates: []
  }

  componentDidMount() {
    this.getRates();
  }

  createCurrency = () => {
    this.props.mutate({ variables: { type: 'NEWC' } }).then((res) => {
      this.getRates();
    })
  }

  getRates = () => {
    client.query({
      query: gql`
        {
          rates(currency: "USD") {
            currency
          }
        }
      `
    }).then(res => {
        this.setState({ rates: res.data.rates.map(r => r.currency) })
    });
  }

  render() {
    const { rates } = this.state;
    return (<div className="container">
      {rates.map(r => <div onClick={this.createCurrency}>{r}</div>)}
    </div>)
  }
}

export default graphql(ADD_TODO)(App);
