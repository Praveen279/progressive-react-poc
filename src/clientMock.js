// import setupClient from 'apollo-client-mock'
import { SchemaLink } from 'apollo-link-schema';
import schema from './mockSchema'
import ApolloClient from "apollo-client";
import { InMemoryCache } from 'apollo-cache-inmemory';

const cache = new InMemoryCache();

export default new ApolloClient({
  link: new SchemaLink({ schema }),
  cache
});
