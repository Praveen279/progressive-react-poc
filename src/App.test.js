import React from 'react';
import ReactDOM from 'react-dom';
import App, {handleChange} from './App';
import {mount} from 'enzyme';
import wait from 'waait';
import { ApolloProvider } from 'react-apollo';
import clientMock from './clientMock';
import './enzyme-adapter';

it('react-test-renderer', () => {
  const tree = mount((
    <ApolloProvider client={clientMock}>
      <App />
    </ApolloProvider>
  ))
  console.log('%%%%%%%%%%%%%%%%%%%%%', tree)
  expect(tree.find('.container').length).toBe(1);
});
