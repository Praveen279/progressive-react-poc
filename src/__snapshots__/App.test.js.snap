// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer 1`] = `undefined`;
